File: cs_manager_system/src/main/java/com/csManager/csmanager/service/VehicleService.java
package com.csManager.csmanager.service;

import java.util.List;
import java.util.Map;

import com.csManager.csmanager.entity.Vehicle;
import com.csManager.csmanager.exception.ShoppingCartNotFoundException;
import com.csManager.csmanager.exception.VehicleNotFoundException;

import org.springframework.http.ResponseEntity;

/**
 * VehicleService
 */
public interface VehicleService {
    ResponseEntity<List<Vehicle>> getAllVehicles();
    ResponseEntity<List<Vehicle>> getAllVehiclesByShoppingCartId(long shoppingCartId) throws ShoppingCartNotFoundException;
    ResponseEntity<Vehicle> getVehicleById(long vehicleId) throws VehicleNotFoundException;
    ResponseEntity<Vehicle> createVehicle(Vehicle vehicle);
    ResponseEntity<Vehicle> updateVehicle(long vehicleId, Vehicle updatedVehicle) throws VehicleNotFoundException;;
    ResponseEntity<Vehicle> partialUpdateVehicle(long vehicleId, Map<Object, Object> fields) throws VehicleNotFoundException;;
    ResponseEntity<?> deleteVehicle(long vehicleId) throws VehicleNotFoundException;;
    ResponseEntity<Vehicle> addVehicleToShoppingCart(long shoppingCartId, long vehicleId) throws Exception;
}

File: cs_manager_frontend/src/service/VehicleService.js
import axios from "axios";

const VEHICLE_BASE_URL = "http://localhost:8080/api/v1/vehicles";

class VehicleService {
    createVehicle(vehicle) {
        return axios.post(VEHICLE_BASE_URL, vehicle)
            .then(response => response.data)
    }

    getAllVehicles() {
        return axios.get(VEHICLE_BASE_URL)
            .then(response => response.data)
    }

    getVehicleById(vehicleId) {
        return axios.get(VEHICLE_BASE_URL + `/${vehicleId}`)
            .then(response => response.data)
    }

    updateVehicle(vehicleId, newVehicle) {
        return axios.put(VEHICLE_BASE_URL + `/${vehicleId}`, newVehicle)
            .then(response => response.data)
    }

    partialUpdateVehicle(vehicleId, newVehicle) {
        return axios.patch(VEHICLE_BASE_URL + `/${vehicleId}`, newVehicle)
            .then(response => response.data)
    }

    deleteVehicle(vehicleId) {
        return axios.delete(VEHICLE_BASE_URL + `/${vehicleId}`)
            .then(response => response.data)
    }

    addVehicleToShoppingCart(vehicleId, shoppingCartId) {
        return axios.post(VEHICLE_BASE_URL + `/${vehicleId}/sc/${shoppingCartId}`, {})
            .then(response => response.data)
    }
}

export default new VehicleService()
